plugin/editorconfig.vim

```diff
@@ -107,50 +117,56 @@ function! s:InitializeVimCore()
     endtry
     return 0
 endfunction
+
+function! s:InitializeExternalCommand()
+" Initialize external_command mode
+
+    if empty(s:editorconfig_exec_path)
+        echo 'Please specify a g:EditorConfig_exec_path'
+        return 1
+    endif
+
+    if g:EditorConfig_verbose
+        echo 'Checking for external command ' . s:editorconfig_exec_path
+    endif
+
+    if !executable(s:editorconfig_exec_path)
+        echo 'File ' . s:editorconfig_exec_path . ' is not executable.'
+        return 1
+    endif
+
+    return 0
+endfunction
 " }}}1
 
 function! s:Initialize() " Initialize the plugin.  {{{1
     " Returns truthy on error, falsy on success.
 
-    " Do some initialization if the user has specified a core mode {{{2
-    if exists('s:editorconfig_core_mode') && !empty(s:editorconfig_core_mode)
+    if empty(s:editorconfig_core_mode)
+        let s:editorconfig_core_mode = 'vim_core'   " Default core choice
+    endif
 
-        if s:editorconfig_core_mode ==? 'vim_core'
-            if s:InitializeVimCore()
-                echo 'EditorConfig: Failed to initialize vim_core mode'
-                return 1
-            endif
-        else
-            echo "EditorConfig: I don't know how to use mode " . s:editorconfig_core_mode
-            return 1
+    if s:editorconfig_core_mode ==? 'external_command'
+        if s:InitializeExternalCommand()
+            echo 'EditorConfig: Failed to initialize external_command mode.  ' .
```

Comment:
Because our default is vim_core mode, likely this is something unexpected to user. Can we do a warning color?

```vim
  echohl WarningMsg
  echo "Warning message"
  echohl None
```